src/gcp/cloudfunctionsv2.ts

```diff
@@ -671,63 +669,78 @@ export function endpointFromFunction(gcfFunction: CloudFunction): backend.Endpoi
-      if (prod === null) {
-        logger.debug("Prod should always return a valid memory amount");
-        return prod as never;
+  if (gcfFunction.serviceConfig) {
+    proto.copyIfPresent(
+      endpoint,
+      gcfFunction.serviceConfig,
+      "ingressSettings",
+      "environmentVariables",
+      "secretEnvironmentVariables",
+      "timeoutSeconds",
+      "uri"
+    );
+    proto.renameIfPresent(
+      endpoint,
+      gcfFunction.serviceConfig,
+      "serviceAccount",
+      "serviceAccountEmail"
+    );
+    proto.convertIfPresent(
+      endpoint,
+      gcfFunction.serviceConfig,
+      "availableMemoryMb",
+      "availableMemory",
+      (prod) => {
+        if (prod === null) {
+          logger.debug("Prod should always return a valid memory amount");
+          return prod as never;
+        }
+        const mem = mebibytes(prod);
+        if (!backend.isValidMemoryOption(mem)) {
+          logger.warn("Converting a function to an endpoint with an invalid memory option", mem);
+        }
+        return mem as backend.MemoryOptions;
       }
-      const mem = mebibytes(prod);
-      if (!backend.isValidMemoryOption(mem)) {
-        logger.warn("Converting a function to an endpoint with an invalid memory option", mem);
+    );
+    proto.convertIfPresent(endpoint, gcfFunction.serviceConfig, "cpu", "availableCpu", (cpu) => {
+      let cpuVal: number | null = Number(cpu);
+      if (Number.isNaN(cpuVal)) {
+        cpuVal = null;
       }
-      return mem as backend.MemoryOptions;
-    }
-  );
-  proto.renameIfPresent(endpoint, gcfFunction.serviceConfig, "minInstances", "minInstanceCount");
-  proto.renameIfPresent(endpoint, gcfFunction.serviceConfig, "maxInstances", "maxInstanceCount");
-  proto.copyIfPresent(endpoint, gcfFunction, "labels");
-  if (gcfFunction.serviceConfig.vpcConnector) {
-    endpoint.vpc = { connector: gcfFunction.serviceConfig.vpcConnector };
+      return cpuVal;
+    });
+    proto.renameIfPresent(endpoint, gcfFunction.serviceConfig, "minInstances", "minInstanceCount");
+    proto.renameIfPresent(endpoint, gcfFunction.serviceConfig, "maxInstances", "maxInstanceCount");
     proto.renameIfPresent(
-      endpoint.vpc,
+      endpoint,
       gcfFunction.serviceConfig,
-      "egressSettings",
-      "vpcConnectorEgressSettings"
+      "concurrency",
+      "maxInstanceRequestConcurrency"
     );
+    proto.copyIfPresent(endpoint, gcfFunction, "labels");
+    if (gcfFunction.serviceConfig.vpcConnector) {
+      endpoint.vpc = { connector: gcfFunction.serviceConfig.vpcConnector };
+      proto.renameIfPresent(
+        endpoint.vpc,
+        gcfFunction.serviceConfig,
+        "egressSettings",
+        "vpcConnectorEgressSettings"
+      );
+    }
+    const serviceName = gcfFunction.serviceConfig.service;
+    if (!serviceName) {
+      logger.debug(
+        "Got a v2 function without a service name." +
+          "Maybe we've migrated to using the v2 API everywhere and missed this code"
```

Comment:
Who is "we" and do we need to surface this to the end user?

Reply:
Ah, I'm seeing this was mentioned in a later line, feel free to ignore too. :+1:

Reply:
We is us the Firebase team. I think this error might be too edge-cases too edge-casey to need to surface to the user (and hopefully not very common).